src/nvim/window.c

```diff
@@ -725,8 +725,11 @@ void win_config_float(win_T *wp, FloatConfig fconfig)
 
   bool has_border = wp->w_floating && wp->w_float_config.border;
   for (int i = 0; i < 4; i++) {
-    wp->w_border_adj[i] =
-      has_border && wp->w_float_config.border_chars[2 * i+1][0];
+    int new_adj = has_border && wp->w_float_config.border_chars[2 * i + 1][0];
+    if (new_adj != wp->w_border_adj[i]) {
+      change_border = true;
```

Comment:
this is not fully general, the border char could change as well. I would replicate the `memcmp` call above for `fconfig.border_chars` as well.

Reply:
Isn't there already `w_redr_border` for that?
<https://github.com/neovim/neovim/blob/1fdbd29dfa6366f8346693d0bf67f4f782ab0f32/src/nvim/window.c#L777-L781>

Reply:
I think you are right :)

Reply:
I just also tried removing that `memcmp` as well. Theoretically it is not necessary any more.

Reply:
though in that case we shouldn't need to do it for `border_hl_ids` as well..

Reply:
That `change_border` name is indeed a bit ambiguous.

Reply:
> I just also tried removing that `memcmp` as well. Theoretically it is not necessary any more.

Test failed. Reverted.

Reply:
ok, let's merge this as-is, if it it works for now. we can get back to cleaning it up later.

Reply:
Oh, that because without that `memcmp` highlight attrs won't be recalculated:
<https://github.com/neovim/neovim/blob/1fdbd29dfa6366f8346693d0bf67f4f782ab0f32/src/nvim/window.c#L741-L744>

Reply:
> ok, let's merge this as-is, if it it works for now. we can get back to cleaning it up later.
@bfredl ping?

Reply:
> ok, let's merge this as-is, if it it works for now. we can get back to cleaning it up later.

@bfredl ping?